### ISSUE TITLE
Editor: Make rail draggable at both ends

### ISSUE THREAD

**Issue by nalquas** [Libre-TrainSim/Libre-TrainSim#425]
### Summary
I find it highly annoying that rails are only draggable at one end. We should look into changing that at some point to improve editor usability.

### Proposal
Allow dragging rails at both ends instead of one.

### Alternatives
None (?)

**Comment by HaSa1002** [comment]
Can we simply flip the rail and reuse the existing code or is it actually easier to create dedicated support for that feature. Considering how we also need to fix the snapping.
CC @DasCapschen

**Comment by DasCapschen** [comment]
The question is: What behaviour do you expect when dragging at the start?
The same behaviour when dragging the end?
Or perhaps just translating the entire rail?

If the former, we could either flip the rail, like HaSa said. Or if you want it without flipping, it will get a lot more complicated.

**Comment by nalquas** [comment]
> The same behaviour when dragging the end?

Yeah, I basically want to be able to drag the rail at both endpoints.

> Or perhaps just translating the entire rail?

No, translating the entire rail can already be done via the gizmo.

Regarding flipping, I think it's fine to flip the rail when dragging at the other end, however I think that when letting go of the rail, the original direction should be applied again.
On that point, it might be a cool feature to be able to flip a rail's direction via a button or something.
